doc/metadata_routing.rst

```diff
@@ -63,28 +80,31 @@ in this section require the following imports and data::
 Weighted scoring and fitting
 ----------------------------
 
-Here :class:`~model_selection.GroupKFold` requests ``groups`` by default. However, we
-need to explicitly request weights for our scorer and the internal cross validation of
-:class:`~linear_model.LogisticRegressionCV`. Both of these *consumers* know how to use
-metadata called ``sample_weight``::
-
-  >>> weighted_acc = make_scorer(accuracy_score).set_score_request(
-  ...     sample_weight=True
-  ... )
-  >>> lr = LogisticRegressionCV(
-  ...     cv=GroupKFold(), scoring=weighted_acc,
-  ... ).set_fit_request(sample_weight=True)
-  >>> cv_results = cross_validate(
-  ...     lr,
-  ...     X,
-  ...     y,
-  ...     params={"sample_weight": my_weights, "groups": my_groups},
-  ...     cv=GroupKFold(),
-  ...     scoring=weighted_acc,
-  ... )
-
-Note that in this example, ``my_weights`` is passed to both the scorer and
-:class:`~linear_model.LogisticRegressionCV`.
+The splitter used internally in :class:`~linear_model.LogisticRegressionCV`,
+:class:`~model_selection.GroupKFold`, requests ``groups`` by default. However, we need
+to explicitly request `sample_weight` for it and for our custom scorer by specifying
+`sample_weight=True` in :class:`~linear_model.LogisticRegressionCV`s `set_fit_request()`
+method and in :func:`~metrics.make_scorer`s `set_score_request()` method. Both
+:term:`consumers <consumer>` know how to use ``sample_weight`` in their `fit()` or
+`score()` methods. We can then pass the metadata in
+:func:`~model_selection.cross_validate` that will route it to any active consumers::
+
+>>> weighted_acc = make_scorer(accuracy_score).set_score_request(sample_weight=True)
+>>> lr = LogisticRegressionCV(
+...     cv=GroupKFold(),
+...     scoring=weighted_acc
+... ).set_fit_request(sample_weight=True)
+>>> cv_results = cross_validate(
+...     lr,
+...     X,
+...     y,
+...     params={"sample_weight": my_weights, "groups": my_groups},
+...     cv=GroupKFold(),
+...     scoring=weighted_acc,
+... )
+
+Note that in this example, :func:`~model_selection.cross_validate` routes ``my_weights``
+to both the scorer and :class:`~linear_model.LogisticRegressionCV`.
```

Comment:
Let's explicitly name the `UnsetMetadataPassedError` then. The next sentence about error handling was already referring to it.

It's not that trivial though. If you only forget to `set_score_request(sample_weight=True)` on the scorer, you won't get an error message (since your `sample_weight` has been used somewhere). There is no way to know where a user intended to use the routed metadata, except for when they forget to use the `set_{method}_requests` entirely. Not routing it to the scorer could be done intentionally.

I tried to put this in words, so that it's simple enough for that introductory section and still conveys how that error works.